src/app/component/places/places.component.ts

```diff
@@ -0,0 +1,254 @@
+import { TranslateService } from '@ngx-translate/core';
+import { Component, OnInit, ViewChild, DoCheck } from '@angular/core';
+import { LocalStorageService } from '@global-service/localstorage/local-storage.service';
+import { HttpClient } from '@angular/common/http';
+import { MatDrawer } from '@angular/material';
+import { AgmMarker } from '@agm/core';
+@Component({
+  selector: 'app-places',
+  templateUrl: './places.component.html',
+  styleUrls: ['./places.component.scss']
+})
+export class PlacesComponent implements OnInit, DoCheck {
+  public lat = 49.84579567734425;
+  public lng = 24.025124653312258;
+  public zoom = 13;
+  public cardsCollection: Array<object>;
+  public tagList: Array<string> = [];
+  public contentObj: any;
+  public favoritePlaces: Array<number> = [];
+  public stars: Array<string> = ['', '', '', '', ''];
+  public markerList: Array<any> = [];
+  public markerListCopy: Array<any> = [];
+  public searchName = '';
+
+  public redIcon = 'assets/img/places/red-marker.png';
+  public greenIcon = 'assets/img/places/green-marker.png';
+  public bookmark = 'assets/img/places/bookmark-default.svg';
+  public bookmarkSaved = 'assets/img/places/bookmark-set.svg';
+  public star = 'assets/img/places/star-1.png';
+  public starHalf = 'assets/img/places/star-filled-half.png';
+  public starUnfilled = 'assets/img/places/star-2.png';
+
+  public apiKey = 'AIzaSyB3xs7Kczo46LFcQRFKPMdrE0lU4qsR_S4';
+
+  @ViewChild('drawer', { static: false }) drawer: MatDrawer;
+
+  constructor(private localStorageService: LocalStorageService, private translate: TranslateService, private http: HttpClient) {}
+
+  ngOnInit() {
+    this.tagList = ['Shops', 'Restaurants', 'Recycling points', 'Events', 'Saved places'];
+
+    this.contentObj = {
+      cardName: '',
+      cardAddress: '',
+      cardText: '',
+      cardImgUrl: '',
+      cardRating: 0,
+      cardStars: []
+    };
+
+    this.cardsCollection = [
```

Comment:
51 - 137 IMHO need be placed in a separate file.

Reply:
Yea, that's right. These rows should be in the service, but I don't have endpoints, and when Zahar makes them, all logic can change, so that will be useless. I'll make this when endpoints arrive.

Reply:
Done